### PR TITLE
Error in assigning exercises as an online assessment component (in session) 3153

### PR DESCRIPTION
**Pull request by carlangas159**
Error in assigning exercises as an online assessment component (in session) 3153

Adjust to establish the session to the generated link.

The problem was that the session was not assigned at the time of creating the link. therefore the session id is assigned before generating the course query, this makes it possible to display the selector of the elements assigned to the session and also to the course when they are active.

### CODE REVIEW

File: main/gradebook/lib/fe/linkform.class.php [hunk 1]
```diff
@@ -96,32 +96,44 @@ protected function build_create()
...
             // and if there are no links left
-            if (!$link->needs_name_and_description() && count($link->get_all_links()) == '0') {
-                $select->addoption($link->get_type_name(), $linkType, 'disabled');
-            } else {
-                $select->addoption($link->get_type_name(), $linkType);
-            }
+            if (!$link->needs_name_and_description() && count($link->get_all_links()) == '0')
+                {
+                    $select->addoption($link->get_type_name(), $linkType, 'disabled');
+                } else
```
Comment (jmontoyaa) [inline main/gradebook/lib/fe/linkform.class.php:116]: Codacy found an issue: [Line indented incorrectly; expected 12 spaces, found 16](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5759046)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5759046&bid=18850882)

File: main/gradebook/lib/fe/linkform.class.php [hunk 2]
```diff
@@ -96,32 +96,44 @@ protected function build_create()
...
 
         foreach ($linkTypes as $linkType) {
             // The hot potatoe link will be added "inside" the exercise option.
-            if ($linkType == LINK_HOTPOTATOES) {
-                continue;
-            }
+            if ($linkType == LINK_HOTPOTATOES)
+                {
+                    continue;
+                }
```
Comment (jmontoyaa) [inline main/gradebook/lib/fe/linkform.class.php:107]: Codacy found an issue: [Line indented incorrectly; expected 12 spaces, found 3](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5759046)

Comment (AngelFQC) [reply main/gradebook/lib/fe/linkform.class.php:107]: Este `if`no debería cambiar. Y el cambio no cumple con las [convenciones de código](https://github.com/chamilo/chamilo-lms/wiki/Coding-conventions).

File: main/gradebook/lib/fe/linkform.class.php [hunk 3]
```diff
@@ -96,32 +96,44 @@ protected function build_create()
...
                 $linkHot = $this->createLink(LINK_HOTPOTATOES, $courseCode);
                 $linkHot->setHp(true);
                 if ($linkHot->get_all_links(true)) {
+
                     $select->addoption(
                         '&nbsp;&nbsp;&nbsp;'.$linkHot->get_type_name(),
                         LINK_HOTPOTATOES
                     );
                 } else {
+
```
Comment (AngelFQC) [inline main/gradebook/lib/fe/linkform.class.php:136]: Espacio en blanco no necesario

**Pull request closed without merging**